content/general/concepts/algorithm/terms/bubble-sort/bubble-sort.md

```diff
@@ -0,0 +1,72 @@
+---
+Title: 'Bubble Sort Algorithm'
+Description: 'Illustrates a simple sorting algorithm using repeated swapping of adjacent elements.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+  - 'Code Foundations'
+Tags:
+  - 'Sorting Algorithms'
+  - 'Algorithms'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Bubble sort** is a simple, comparison-based [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that repeatedly steps through an [array](https://www.codecademy.com/resources/docs/java/arrays), comparing and swapping adjacent elements if they are in the wrong order. This process continues until the array is sorted, making it an intuitive method for sorting arrays.
+
+## BubbleSort Method
+
+**Process**:
+
+- `Iterative Comparison`: Start with the first element and compare it with the next element.
+- `Swap`: If the current element is greater than the next element, swap them to reorder.
+- `Repeat`: Continue this process for each element in the array, iterating multiple times over the entire array.
+
+**Completion**:
+
+- `End Condition`: The array is sorted when a complete pass occurs without any swaps.
```

Comment:
You can write the steps of the algorithm within this heading directly so it gives a better idea about the algorithm. Check out this [entry](https://www.codecademy.com/resources/docs/general/algorithm/binary-search) for reference.